pkgs/os-specific/linux/nfs-utils/default.nix

```diff
@@ -64,6 +65,13 @@ in stdenv.mkDerivation rec {
         -e "s,/sbin/modprobe,${kmod}/bin/modprobe,g" \
         -e "s,/usr/sbin,$out/bin,g" \
         $out/etc/systemd/system/*
+
+      # Fix path to keytab, the existence of which is used to determine whether
+      # or not rpc-gssd.service and auth-rpcgss-module.service should be
+      # started.
+      sed -i -E "s,($out)?/etc/krb5.keytab,${keytabPath},g" \
+        $out/etc/systemd/system/rpc-gssd.service \
+        $out/etc/systemd/system/auth-rpcgss-module.service \
```

Comment:
it is also possible to just override default values in the service in module itself, example:

```
# the first empty string will reset ConditionsExists, so it can be assigned a new value
systemd.services.rpc-gssd.unitConfig.ConditionExists = [ "" 
 "/etc/krb5.keytab"
];
```

That way recompilation is not necessary. This is untested and only based on past experience. Let me know, if it does not work.

Reply:
@Mic92 thanks, I'll give that a try!

Reply:
Unfortunately it doesn't work:

```
eqyiel@ayanami ~ % systemctl status rpc-gssd
● rpc-gssd.service - RPC security service for NFS client and server
   Loaded: loaded (/nix/store/i50nrgfy858ni0lnmr2my8hfi44ybhp0-nfs-utils-2.1.1/etc/systemd/system/rpc-gssd.service; enabled; vendor preset: enabled)
  Drop-In: /nix/store/ds1y652m46301r7xck8ls3d527whi723-system-units/rpc-gssd.service.d
           └─overrides.conf
   Active: inactive (dead) since Wed 2017-09-20 20:13:19 ACST; 2min 46s ago
Condition: start condition failed at Wed 2017-09-20 20:14:30 ACST; 1min 35s ago
           └─ ConditionPathExists=/nix/store/i50nrgfy858ni0lnmr2my8hfi44ybhp0-nfs-utils-2.1.1/etc/krb5.keytab was not met
 Main PID: 747 (code=exited, status=0/SUCCESS)
```

That override does contain the correct path, but it doesn't seem to be used:
```
eqyiel@ayanami ~ % cat /nix/store/ds1y652m46301r7xck8ls3d527whi723-system-units/rpc-gssd.service.d/overrides.conf
[Unit]
ConditionPathExists=/etc/krb5.keytab
X-Restart-Triggers=/nix/store/5pgzyknq6bf809003kc5v0qcxqz7gzlg-nfs.conf

[Service]
Environment="LOCALE_ARCHIVE=/nix/store/x36zk6cx0xi71593vkk4wski2vbg3sln-glibc-locales-2.25-49/lib/locale/locale-archive"
Environment="PATH=/nix/store/v3mhywkcj5vycgclmf2j8zlpbayvxndk-coreutils-8.28/bin:/nix/store/simjpbiksmvvg2xa25v65g5hk2sr014n-findutils-4.6.0/bin:/nix/store/dlk67w96zmbh4gxi72vkxdh4pm8xdb7j-gnugrep-3.1/bin:/nix/store/c4halcrqi0dbjjd6i9av69z5brgav3pz-gnused-4.4/bin:/nix/store/xk66kbaqg4l72xpyxprwhm35dz8mzm1d-systemd-234/bin:/nix/store/v3mhywkcj5vycgclmf2j8zlpbayvxndk-coreutils-8.28/sbin:/nix/store/simjpbiksmvvg2xa25v65g5hk2sr014n-findutils-4.6.0/sbin:/nix/store/dlk67w96zmbh4gxi72vkxdh4pm8xdb7j-gnugrep-3.1/sbin:/nix/store/c4halcrqi0dbjjd6i9av69z5brgav3pz-gnused-4.4/sbin:/nix/store/xk66kbaqg4l72xpyxprwhm35dz8mzm1d-systemd-234/sbin"
Environment="TZDIR=/nix/store/ghziaza6fy9x74b9pnlc4zv8bc4cy8p2-tzdata-2016j/share/zoneinfo"
```

Reply:
I would much prefer to configure it this way if it were possible though.